streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java

```diff
@@ -428,6 +428,68 @@ public void shouldUseCorrectDefaultsWhenNoneSpecified() {
         assertTrue(config.defaultTimestampExtractor() instanceof FailOnInvalidTimestamp);
     }
 
+    @Test(expected = StreamsException.class)
+    public void shouldSpecifyCorrectKeySerdeClassOnErrorUsingDeprecatedConfigs() {
+        final Properties props = minimalStreamsConfig();
+        props.put(StreamsConfig.KEY_SERDE_CLASS_CONFIG, MisconfiguredSerde.class);
+        try {
+            final StreamsConfig config = new StreamsConfig(props);
```

Comment:
Nit: Move this line before the try-catch-block. The try-catch-block should only contain the stuff we want to test.